pkgs/desktops/gnome/misc/gnome-flashback/default.nix

```diff
@@ -180,10 +180,13 @@ let
             dontWrapGApps = true; # We want to do the wrapping ourselves.
 
             # gnome-flashback and gnome-panel need to be added to XDG_DATA_DIRS so that their .desktop files can be found by gnome-session.
+            # We need to pass the --builtin flag so that gnome-session invokes gnome-session-binary instead of systemd.
+            # If systemd is used, it doesn't use the environment we set up here and so it can't find the .desktop files.
             preFixup = ''
               makeWrapper ${gnome-session}/bin/gnome-session $out \
-                --add-flags "--session=gnome-flashback-${wmName}" \
+                --add-flags "--session=gnome-flashback-${wmName} --builtin" \
                 --set-default XDG_CURRENT_DESKTOP 'GNOME-Flashback:GNOME' \
+                --prefix PATH : '${lib.makeSearchPath "bin" [ ibus ]}' \
                 --prefix XDG_DATA_DIRS : '${lib.makeSearchPath "share" ([ wmApplication gnomeSession gnome-flashback ] ++ lib.optional enableGnomePanel gnome-panel)}' \
```

Comment:
Something like this might work.

```nix
systemd.user.services.gnome-flashback = {
  environment = {
    ...
  }
}
```

Though it is not clear to me what the problem is – there does not seem to be any significant difference in gnome-flashback-3.44.0…3.46.0 service files or in gnome-session-42.0…43.0.